pkg/transport/kafka/kafka_consumer_client.go

```diff
@@ -0,0 +1,292 @@
+package kafka
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/go-logr/logr"
+	datatypes "github.com/open-cluster-management/hub-of-hubs-data-types"
+	kafkaconsumer "github.com/open-cluster-management/hub-of-hubs-kafka-transport/kafka-client/kafka-consumer"
+	kafkaHeaderTypes "github.com/open-cluster-management/hub-of-hubs-kafka-transport/types"
+	compressor "github.com/open-cluster-management/hub-of-hubs-message-compression"
+	"github.com/open-cluster-management/hub-of-hubs-message-compression/compressors"
+	"github.com/open-cluster-management/leaf-hub-spec-sync/pkg/bundle"
+	"github.com/open-cluster-management/leaf-hub-spec-sync/pkg/transport"
+)
+
+const (
+	commitRescheduleDelay = time.Second * 20
+	bufferedChannelSize   = 42
+)
+
+var (
+	errReceivedUnsupportedBundleType = errors.New("received unsupported message type")
+	errMissingCompressionType        = errors.New("compression type is missing from message headers")
+)
+
+// NewConsumer creates a new instance of Consumer.
+func NewConsumer(log logr.Logger, bundleUpdatesChan chan *bundle.ObjectsBundle) (*Consumer, error) {
+	kafkaConfigMap, topics, err := getKafkaConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create consumer: %w", err)
+	}
+
+	msgChan := make(chan *kafka.Message, bufferedChannelSize)
+
+	kafkaConsumer, err := kafkaconsumer.NewKafkaConsumer(kafkaConfigMap, msgChan, log)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create consumer: %w", err)
+	}
+
+	if err := kafkaConsumer.Subscribe(topics); err != nil {
+		return nil, fmt.Errorf("failed to create consumer: %w", err)
```

Comment:
change error to `failed to subscribe to requested topics` + add `kafkaConsumer.Close()` to avoid memory leak.